Breeze/UserDefaults.swift

```diff
@@ -211,6 +229,100 @@ extension UserDefaults{
         set(0, forKey: UserDefaultsKeys.currentIsland.rawValue)
     }
     
+    func checkDayRollover() {
+        // if lastTimeProtectedDataStatusChecked is previous day, then store previous day's time and send to Sendgrid??
+        let calendar = Calendar.current
+        let prevDate = Date(timeIntervalSince1970: double(forKey: UserDefaultsKeys.lastTimeProtectedDataStatusChecked.rawValue))
+        let prevComponents = calendar.dateComponents([.weekday], from: prevDate)
+        let prevDayOfWeek = prevComponents.weekday
+        
+        let currDate = Date()
+        let currComponents = calendar.dateComponents([.weekday], from: currDate)
+        let currDayOfWeek = currComponents.weekday
+        
+        //If it is a new day, then reset the current phone usage and add it to statistics and send to Sendgrid
+        if (prevDayOfWeek != currDayOfWeek) {
+            resetCurrentPhoneUsage()
```

Comment:
maybe call addIntervalToCurrentPhoneUsage() so currentDayPhoneUsage updated before resetting currentPhoneUsage?

Reply:
Or actually we might need a check to see if protecteddata was available during the last check and then potentially also add the difference between time last check and midnight if the date has changed. and then add the remaining to the total for the new day.

Reply:
Agreed, I'm just hoping the intervals get small enough but either way tbh it shouldn't matter too much

Reply:
Actually I think what is going to happen is: let's say you are on your phone at 11:00pm one day, and then on your phone at 1:00am the next day, then currentPhoneUsage will reset, and then add the 2 hours. So the interval gets tagged onto the next day.